### ISSUE TITLE
Is the dedicated server currently working?

### ISSUE THREAD

**Issue by wafflecomposite** [Jan4V/hlnonvr#3]
Hi, I'm trying to test the dedicated server locally. If it's supposed to be working now, I'm probably doing something wrong, can you help?

I'm running hlnonvr v0.5, not sure how to get HLA version, but the one that is currently the newest.
How I assume it should work - I'm launching hlnonvr.exe with `-dedicated -allow_no_lobby_connect` args, then i choose a map there, for example `map a1_intro_world`, it seems to be loading correctly.

Then I launch a separate HLA instance just with  `-console -vconsole` args and trying to `connect localhost`. Almost immediately the server goes down with an error, seems to be 0xC0000005, I'm not sure how to get the logs, but I found and attached a dump file there

[hlnonvr_accessviolation_mdmp.zip](https://github.com/Jan4V/hlnonvr/files/4464779/hlnonvr_accessviolation_mdmp.zip)

In any case, thank you for your work!

**Comment by sabi1** [comment]
It appears that the access violation is being caused by the Server.dll trying to access something in the adress 0xFFFFFFFFFFFFFFFF which it doesnt have rights to. The problem would be that the server.dll had to be modified and debugged, which is hardly possible without the server.pdb file, so the only way would be trying to create an alternative dll like you see in the hlnonvr.dll

**Comment by InsDel2113** [comment]
MP and Dedicated servers have been broken for awhile. Valve patched out or broke the code required to due it, therefore the patch no longer works.

**Comment by sampavlovic** [comment]
Are there any plans to fix it or any pointers on what part of the code it was so I can try to identify the current offset with Ghidra?

**Comment by InsDel2113** [comment]
> Are there any plans to fix it or any pointers on what part of the code it was so I can try to identify the current offset with Ghidra?

Sorry for the late reply

It wouldn't be easy to fix (at least I don't think it is)
I'm currently on a vacation and away from home, however when I'm back I could probably do some comparison and find out whats broken and what changed between the last working MP version -> the one we have now, which is broken.

I've experimented a lot in X64DBG, nopping things out, changing values, etc. That's how I got MP going originally, after that, Jan4V added it to his patch and did it a little more proper than my patch. My REing skills are still very meh.

I'll be doing lots of work on it when I'm home, thanks for the interest!
